src/Tools/ExternalAccess/FSharp/Internal/Classification/FSharpClassificationService.cs

```diff
@@ -48,5 +47,15 @@ public ClassifiedSpan AdjustStaleClassification(SourceText text, ClassifiedSpan
         {
             return _service.AdjustStaleClassification(text, classifiedSpan);
         }
+
+        public Task<object?> GetDataToCacheAsync(Document document, CancellationToken cancellationToken)
+        {
+            return SpecializedTasks.Default<object?>();
+        }
+
+        public Task<TextChangeRange?> ComputeSyntacticChangeRangeAsync(Document oldDocument, Document newDocument, CancellationToken cancellationToken)
```

Comment:
```suggestion
        public ValueTask<TextChangeRange?> ComputeSyntacticChangeRangeAsync(Document oldDocument, Document newDocument, CancellationToken cancellationToken)
```

Reply:
annoying!  if i do that, then .WithTimeout becomes not available (it's a Task<T> extension).  SO i'd have to wrap with .AsTask().  Is that ok/ideal?  Or does that negate ValueTask here?

Reply:
I wouldn't worry about it. The asynchrony will go away once the best-within-time strategy is implemented.